contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -799,37 +804,39 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
         uint256 balanceDiff
     ) internal {
         address tokenAddressIn = _getDestinationBaseAsset();
-        (address[] memory converterAddresses, uint256[] memory converterBalances) = converterNetwork
-        .findTokenConvertersForConverters(tokenAddressOut, tokenAddressIn);
         uint256 convertedTokenOutBalance = balanceDiff;
         uint256 convertedTokenInBalance;
-        uint256 convertersLength = converterAddresses.length;
-        for (uint256 i; i < convertersLength; ) {
-            uint256 amountOutMantissa = converterBalances[i];
-            if (amountOutMantissa == 0) break;
-            (, uint256 amountIn) = IAbstractTokenConverter(converterAddresses[i]).getUpdatedAmountIn(
-                amountOutMantissa,
-                tokenAddressOut,
-                tokenAddressIn
-            );
-            if (amountIn > convertedTokenOutBalance) {
-                amountIn = convertedTokenOutBalance;
-            }
-            IERC20Upgradeable(tokenAddressOut).approve(converterAddresses[i], amountIn);
-            (, uint256 actualAmountOut) = IAbstractTokenConverter(converterAddresses[i]).convertExactTokens(
-                amountIn,
-                0,
-                tokenAddressOut,
-                tokenAddressIn,
-                destinationAddress
-            );
-            convertedTokenOutBalance -= amountIn;
-            convertedTokenInBalance += actualAmountOut;
-
-            if (convertedTokenOutBalance == 0) break;
-
-            unchecked {
-                ++i;
+        if (address(converterNetwork) != address(0)) {
```

Comment:
This is ok, but early return (`if (address(converterNetwork) == address(0)) return;`) increases the readability of the code, reducing the number of nested blocks